src/NUnitEngine/nunit.engine/Internal/MetadataFacts.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using NUnit.Common;
+using NUnit.Engine.Internal.Metadata;
+
+namespace NUnit.Engine.Internal
+{
+    internal static class MetadataFacts
+    {
+        public static Version ParseMetadataVersion(string metadataVersion)
+        {
+            Guard.ArgumentNotNullOrEmpty(metadataVersion, nameof(metadataVersion));
+
+            var version = metadataVersion;
+
+            if (version.StartsWith("v", StringComparison.OrdinalIgnoreCase))
+                version = version.Substring(1);
+
+            return new Version(version);
+        }
+
+        public static string GetTargetFramework(IAssemblyMetadataProvider assembly)
+        {
+            Guard.ArgumentNotNull(assembly, nameof(assembly));
+
+            if (assembly.GetAttributes("System.Runtime.Versioning.TargetFrameworkAttribute").TryFirst(out var attr))
+            {
+                if (attr.PositionalArguments.FirstOrDefault() is string frameworkName)
+                    return frameworkName;
+            }
+
+            foreach (var reference in assembly.AssemblyReferences)
+            {
+                if (reference.Name == "mscorlib" && BitConverter.ToUInt64(reference.GetPublicKeyToken(), 0) == 0xac22333d05b89d96)
+                {
+                    // We assume 3.5, since that's all we are supporting
+                    // Could be extended to other versions if necessary
+                    // Format for FrameworkName is invented - it is not
+                    // known if any compilers supporting CF use the attribute
+                    return ".NETCompactFramework,Version=3.5";
+                }
```

Comment:
@ChrisMaddock Should this CF thing be removed?

Reply:
Not too sure what this is replacing at this level. The only CF functionality I left in, is the stuff that identifies an assembly which targets CF - in order to return an appropriate error message about that platform not being supported by the console. Would you mind following up the stack, to see where this is called?

I'm away at the moment, but will try and have a proper look at this at some point!

Reply:
When the engine tries to identify target platforms, it relies first of all on the target framework attribute. But the `TargetFrameworkAttribute` was only introduced in (relatively) recent compilers and NUnit has to respond sensibly even if the attribute is not present.

For CF, which was last supported by VS2008, there is no attribute. So I added this code to detect a CF assembly and return a fake value, using a format similar to what is used in the attribute itself. The return is supposed to be used precisely as you suggest: to allow us to give an intelligent error message.